File: app/src/main/java/pro/butovanton/mega/FirstFragment.java
package pro.butovanton.mega;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ProgressBar;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProvider;
import androidx.navigation.fragment.NavHostFragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import pro.butovanton.mega.R;

public class FirstFragment extends Fragment implements ItemClickListener {

    private RecyclerView recyclerView;
    private RecyclerAdapterMega adapter;
    private LinearLayoutManager lm;
    private ProgressBar progressBar;

    private ViewModelModel viewModelModel;

    @Override
    public View onCreateView(
            LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState
    ) {
        return inflater.inflate(R.layout.fragment_first, container, false);
    }

    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);

        progressBar = view.findViewById(R.id.progressBar);

        recyclerView = view.findViewById(R.id.reciclerView);
        adapter = new RecyclerAdapterMega(this, getContext());
        lm = new LinearLayoutManager(getContext());
        recyclerView.setLayoutManager(lm);

        recyclerView.setAdapter(adapter);

        viewModelModel = new ViewModelProvider(this).get(ViewModelModel.class);
        viewModelModel.getModel().observe(getViewLifecycleOwner(), new Observer<List<MModel>>() {
            @Override
            public void onChanged(List<MModel> mModels) {

            }
        });
    }

    @Override
    public void onItemClick(String id) {
        Bundle bundle = new Bundle();
        bundle.putString("id", id);
        NavHostFragment.findNavController(this)
                .navigate(R.id.action_FirstFragment_to_SecondFragment, bundle);
    }
}

File: app/src/main/java/pro/butovanton/mega/FirstFragment.kt
package pro.butovanton.mega

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.navigation.fragment.NavHostFragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

class FirstFragment : Fragment(), ItemClickListener {
    private lateinit var recyclerView: RecyclerView
    private var adapter: RecyclerAdapterMega? = null
    private var lm: LinearLayoutManager? = null
    private var viewModelModel: ViewModelModel? = null
    override fun onCreateView(
            inflater: LayoutInflater, container: ViewGroup?,
            savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_first, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        recyclerView = view.findViewById(R.id.reciclerView)
        adapter = RecyclerAdapterMega(this, context)
        lm = LinearLayoutManager(context)
        recyclerView.layoutManager = lm
        recyclerView.setAdapter(adapter)
        viewModelModel = ViewModelProvider(this).get(ViewModelModel::class.java)
        viewModelModel!!.getModel().observe(viewLifecycleOwner, Observer { models -> adapter!!.set(models) })
    }

    override fun onItemClick(id: String?) {
        Log.d("DEBUG", "click $id")
        val bundle = Bundle()
        bundle.putString("id", id)
        NavHostFragment.findNavController(this)
                .navigate(R.id.action_FirstFragment_to_SecondFragment, bundle)
    }
}